File: c#/Structural/Facade/Publisher/PublisherFacade.cs
using Interface.IArticle;
using Audit.Logger;
using Publisher.Seo;
using Publisher.Slug;
using Publisher.Storage;


namespace Publisher.PublisherFacade
{
    public class ArticlePublisher
    {
        private IArticle article;
        private string user;

        public ArticlePublisher(IArticle article, string user)
        {
            this.article = article;
            this.user = user;
        }

        public async Task Publish()
        {
            var seo = new SeoSuggestions();
            var slugger = new Slugger();
            var logger = new Logger();

            if(!Validator.Validator.Validate(article))
            {
                Console.WriteLine("❌ Publicación cancelada: Artículo inválido");
                return;
            }

            article.Slug = slugger.SlugGenerator(article.Title);
            _ = seo.SeoOptimizer(article);

            article.Html = HTMLConverter.HTMLConverter.PlainHtml(article.Content ?? "");

            await StorageDB.SaveToDB(article);
            StorageDB.PublishToCDN(article);

            logger.LoggerAction(user, "publish");
        }
    }
}

File: typescript/src/structural/facade/publisher/PublisherFacade.ts
import { Article } from "../interfaces/IArticle";
import { validator } from "./Validator";
import { slugGenerator } from "./SlugGenerator";
import { seoOptimizer } from "./SeoOptimizer";
import { plainToHtml } from "./HtmlConverter";
import { saveToDb } from "./Storage";
import { loggerAction } from "../audit/Logger";

export class ArticlePublisher {
    private article: Article;
    private user: string;

    constructor(article: Article, user: string){
        this.article = article
        this.user = user
    }

    public publish(): void {
        validator(this.article);
        this.article.slug = slugGenerator(this.article.title);
        seoOptimizer(this.article);
        this.article.html = plainToHtml(this.article.content || "");
        saveToDb(this.article);
        loggerAction(this.user, "publish");
    }
}